main/src/com/google/refine/commands/OpenExtensionsDirCommand.java

```diff
@@ -0,0 +1,49 @@
+
+package com.google.refine.commands;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.refine.ProjectManager;
+import com.google.refine.io.FileProjectManager;
+
+public class OpenExtensionsDirCommand extends Command {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        if (!hasValidCSRFToken(request)) {
+            respondCSRFError(response);
+            return;
+        }
+
+        String serverName = request.getServerName();
+
+        if (!"127.0.0.1".equals(serverName) && !"localhost".equals(serverName)) {
+            respond(response,
+                    "{ \"code\" : \"error\", \"message\" : \"Extension directory can only be opened on the local machine where OpenRefine is run.\" }");
+        } else if (ProjectManager.singleton instanceof FileProjectManager) {
+            File dir = new File(((FileProjectManager) ProjectManager.singleton).getWorkspaceDir(), "extensions");
+
+            if (Desktop.isDesktopSupported()) {
+                Desktop desktop = Desktop.getDesktop();
+                desktop.open(dir);
+            } else /* if Mac */ {
+                Runtime.getRuntime().exec(
+                        "open .",
+                        new String[] {},
+                        dir);
+            }
```

Comment:
Following up on @magdmartin's comment in #7010, perhaps it would be worth having this logic in a separate function, which would be used both by the "open workspace directory" and the "open extensions directory" features. This way we could refine the support for each platform in a single place. (For instance a utility class with a static method on it.) I'm not suggesting that refinement should be done in this PR, it can be done separately.